src/components/pages/devices/flyouts/deviceNew/deviceNew.js

```diff
@@ -0,0 +1,411 @@
+                </FormGroup>,
+                <FormGroup key="deviceModel">
+                  <FormLabel>{t('devices.flyouts.new.deviceModel.label')}</FormLabel>
+                  <div className="device-model-temp">{t('devices.flyouts.new.deviceModel.hint')} -- TODO: Add options</div>
+                </FormGroup>
+              ]
+            }
+            {
+              !isSimulatedDevice && [
+                <FormGroup key="deviceCount">
+                  <FormLabel>{t('devices.flyouts.new.count.label')}</FormLabel>
+                  <div className="device-count">{this.countLink.value}</div>
+                </FormGroup>,
+                <FormGroup key="deviceId">
+                  <FormLabel>{t('devices.flyouts.new.deviceId.label')}</FormLabel>
+                  <Radio link={this.isGenerateIdLink} key={DeviceIdTypeOptions.manual.hintName} value={DeviceIdTypeOptions.manual.value}>
+                    <FormControl className="device-id" link={this.deviceIdLink} disabled={isGenerateId} type="text" placeholder={t(DeviceIdTypeOptions.manual.hintName)} />
+                  </Radio>
+                  <Radio link={this.isGenerateIdLink} key={DeviceIdTypeOptions.generate.labelName} value={DeviceIdTypeOptions.generate.value}>
+                    {t(DeviceIdTypeOptions.generate.labelName)}
+                  </Radio>
+                </FormGroup>,
+                <FormGroup key="authType">
+                  <FormLabel key={AuthTypeOptions.labelName}>{t(AuthTypeOptions.labelName)}</FormLabel>
+                  <Radio link={this.authenticationTypeLink} key={AuthTypeOptions.symmetric.labelName} value={AuthTypeOptions.symmetric.value}>
+                    {t(AuthTypeOptions.symmetric.labelName)}
+                  </Radio>
+                  <Radio link={this.authenticationTypeLink} key={AuthTypeOptions.x509.labelName} value={AuthTypeOptions.x509.value}>
+                    {t(AuthTypeOptions.x509.labelName)}
+                  </Radio>
+                </FormGroup>,
+                <FormGroup key="authKeyType">
+                  <FormLabel key={AuthKeyTypeOptions.labelName}>{t(AuthKeyTypeOptions.labelName)}</FormLabel>
+                  <Radio link={this.isGenerateKeysLink} key={AuthKeyTypeOptions.generate.labelName} value={AuthKeyTypeOptions.generate.value} disabled={isX509}>
+                    {t(AuthKeyTypeOptions.generate.labelName)}
+                  </Radio>
+                  <Radio link={this.isGenerateKeysLink} key={AuthKeyTypeOptions.manual.labelName} value={AuthKeyTypeOptions.manual.value}>
+                    {t(AuthKeyTypeOptions.manual.labelName)}
+                  </Radio>
+                  <div className="sub-settings">
+                    <FormLabel>{isX509 ? t('devices.flyouts.new.authenticationKey.primaryThumbprint') : t('devices.flyouts.new.authenticationKey.primaryKey')}</FormLabel>
```

Comment:
the label and control should be wrapped in a FormGroup. It will generate and id to make the label work correctly on click